### ISSUE TITLE
Unable to disable the `Layout/CaseIndentation` RuboCop linter.

### ISSUE THREAD

**Issue by mockdeep** [sds/haml-lint#407]
For some reason after upgrading to version 0.46.0, we're unable to disable the `Layout/CaseIndentation` lint. We get a lot of reports, but when I add it to the our custom RuboCop overrides, it won't disable it. That appears to be the only rule that won't disable, though. Others work as intended.

**Comment by MaxLap** [comment]
Because of how the new Rubocop system works, some cops are forced in some specific configuration.

This is needed because the way the system works, haml-lint basically converts the HAML to valid ruby, runs rubocop, and then transfers the fixes. Some cops do transformation that ruin the expected structure of the code. CaseIndentation is one such example.

I'm curious, are the errors you mention in regular haml (with - and =) or in a :ruby filter?

You can see all of the forced rubocop configurations, as well as explanations, here:
https://github.com/sds/haml-lint/blob/main/config/forced_rubocop_config.yml

**Comment by mockdeep** [comment]
@MaxLap the problem is that we have nested `case` statements. E.g.:

```haml
    - case sorting.direction
      - when 'asc'
        = icon('caret-up', aria: { label: 'ascending order' }, role: 'img')
      - when 'desc'
        = icon('caret-down', aria: { label: 'descending order' }, role: 'img')
```

I'm going to try unindenting the blocks, but I'm a little concerned about ambiguity with following blocks:

```haml
- case something
- when 'foo'
- when 'bar'
- unless something_else # this looks like it's part of the `case` even though it's completely separate.
```

**Comment by MaxLap** [comment]
I understand your concern. The effect is less pronounced when you have content in your `when`, such as:

```
- case something
- when 'foo'
  abc
- when 'bar'
  def
- unless something_els
```

The main reason we need to use the non-indented configuration is that HAML fails on this:

```
- case 1
  - when 1
  - when 2
    foo
```

For some reason, it inserts an extra `end` in the generated Ruby code, which is a SyntaxError. So in "indented when" mode, auto-correction could lead to this "valid" HAML that HAML fails to process, which I want to avoid as much as possible.

**Comment by wilfison** [comment]
I dug into this and here's what's going on.

The forced RuboCop config (`config/forced_rubocop_config.yml`) sets:

```yaml
Layout/CaseIndentation:
  Enabled: true
  EnforcedStyle: end
  IndentOneStep: false
```

These forced settings are merged so that they always take precedence over your own `.rubocop.yml` (haml-lint builds an internal config that `inherit_from`s yours, with the forced values on top). That's why adding `Layout/CaseIndentation: Enabled: false` to your RuboCop config has no effect, it gets overridden. As @MaxLap explained, the cop is forced to keep auto-correct from producing HAML that HAML itself can't parse (indented `when` generates an extra `end` in the compiled Ruby).

The thing that _does_ work is haml-lint's own `ignored_cops` option, in your `.haml-lint.yml`:

```yaml
linters:
  RuboCop:
    ignored_cops:
      - Layout/CaseIndentation
```

This passes the cop to RuboCop via `--except`, so it's skipped entirely (both when reporting and when auto-correcting) regardless of the forced config.

So this is really a discoverability problem more than a bug: the escape hatch already exists, but it wasn't documented well (the README even had a typo, `ignored_cop`). Happy to open a small PR that fixes the typo and documents `ignored_cops` as the supported way to disable a forced cop for HAML files.

**Comment by MaxLap** [comment]
The fact that the forced config puts in an `Enabled: true` makes me think there might be a problem with just disabling it, that the cop is required. But this is too far in the past for me to remember.
* You could try running the specs with it forced disabled. If they pass, then it's likely ok.
* I also wonder if more recent versions of HAML could have solved the problem since. I think HAML felt quite dead when I developed this so I didn't bother with opening an issue, but things have evolved since. There would be a bit of a backward compat issue in changing it, but it would be interesting to know.
